### ISSUE TITLE
Support different stop strategies and enable time_budget_s for ludwig train

### ISSUE THREAD

**Issue by Jeffwan** [ludwig-ai/ludwig#2541]
**Is your feature request related to a problem? Please describe.**

Currently, I think the stop condition for ludwig train job is if there's no more improvement after a few steps, then it's stopped. 
Sometimes, I want to stop my job after xx minutes/hrs. Current policy is not flexible enough.

**Describe the use case**

As an end user, I hope I have multiple ways to stop the job or use a combined strategy.  To be concrete, I need `time_budget_s` in `ludwig train` as well, currently, it's just available in hypopt because underneath `ray.tune` provides it. 

**Describe the solution you'd like**

Have an abstraction on Stop Condition or Policy.  Accuracy based or  Time based could be one of the implementations.  

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by tgaddair** [comment]
Hey @Jeffwan, we can definitely come up with a solution for this in the config. If you're looking for a programmatic workaroumd, you can achieve this using `ludwig.callbacks.Callback`:

```
start_time = time.time()
time_budget_s = 3600

class EarlyStopCallback(Callback):
    def should_early_stop(self, trainer, progress_tracker, is_coordinator):
        # Triggers early stopping if any callback on any worker returns True
        return time.time() - start_time > time_budget_s

model = LudwigModel(..., callbacks=[EarlyStopCallback()])
model.train(...)
```

Let me know if this works to unblock you for the short-term.

**Comment by Jeffwan** [comment]
I use train_cli method and notice it accepts the callbacks as well. Thanks! I verified it works for me. 

```
from ludwig.train import train_cli
train_cli(.., callbacks=[EarlyStopCallback()])
```
